spec/rubocop/cop/rspec/repeated_subject_call_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::RepeatedSubjectCall do
+  it 'registers an offense for a singular block' do
+    expect_offense(<<-RUBY)
+      RSpec.describe Foo do
+        it do
+          subject
+          expect { subject }.to not_change { Foo.count }
+          ^^^^^^^^^^^^^^^^^^ Calls to subject are memoized, this block is misleading
+        end
+      end
+    RUBY
+  end
+
+  it 'registers an offense for repeated blocks' do
+    expect_offense(<<-RUBY)
+      RSpec.describe Foo do
+        it do
+          expect { subject }.to change { Foo.count }
+          expect { subject }.to not_change { Foo.count }
+          ^^^^^^^^^^^^^^^^^^ Calls to subject are memoized, this block is misleading
+        end
+      end
+    RUBY
+  end
+
+  it 'registers an offense for nested blocks' do
+    expect_offense(<<-RUBY)
+      RSpec.describe Foo do
+        it do
+          expect(subject.a).to eq(3)
+          nested_block do
+            expect { on_shard(:europe) { subject } }.to not_change { Foo.count }
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Calls to subject are memoized, this block is misleading
+          end
+        end
+      end
+    RUBY
+  end
+
+  it 'registers an offense for custom subjects' do
+    expect_offense(<<-RUBY)
+      RSpec.describe Foo do
+        subject(:bar) { do_something }
+
+        it do
+          bar
+          expect { bar }.to not_change { Foo.count }
+          ^^^^^^^^^^^^^^ Calls to subject are memoized, this block is misleading
+        end
+      end
+    RUBY
+  end
+
+  it 'registers no offenses for no block' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe Foo do
+        it do
+          expect(subject.a).to eq(3)
+          expect(subject.b).to eq(4)
+        end
+      end
+    RUBY
+  end
+
+  it 'registers no offenses for block first' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe Foo do
+        it do
+          expect { subject }.to change { Foo.count }
+          expect(subject.b).to eq(4)
+        end
+      end
+    RUBY
```

Comment:
I would be interested in seeing a similar example, but with two `it` blocks:
```rb
RSpec.describe Foo do
  it do
    expect { subject }.to change { Foo.count }
  end

  it do
    expect(subject.b).to eq(4)
  end
end
```
In this case, the first `subject` block _should_ cause an offense, right?

Reply:
Added a spec for this - no offense should be raised since those are unique example blocks and `subject` is evaluated once for each.

Reply:
Huh, I had somehow got the impression that `let` and `subject` memoized calls not per example, but per example group. I was wrong 😬

I think the added example still has value though.